tests/run-pass/non_extern_foreign_call.rs

```diff
@@ -0,0 +1,39 @@
+#![feature(rustc_attrs)]
+
+#[no_mangle]
+extern "C" fn foo() -> i32 {
+    -1
+}
+
+#[export_name = "bar"]
+fn bar() -> i32 {
+    -2
+}
+
+#[rustc_std_internal_symbol]
+fn baz() -> i32 {
+    -3
+}
+
+#[no_mangle]
+extern "C" fn exit(_: i32) -> ! {
+    unreachable!()
+}
+
+#[no_mangle]
+extern "C" fn ExitProcess(_: u32) -> ! {
+    unreachable!()
+}
```

Comment:
What is this about? Looks like a leftover from when you supported overwriting shims.

Reply:
This is used to make sure custom `#[no_mangle]` functions can't override shims.